test/jsonschema/jsonschema_frame_2020_12_test.cc

```diff
@@ -665,3 +680,74 @@ TEST(JSONSchema_frame_2020_12, explicit_argument_id_different) {
 
   EXPECT_TRUE(references.empty());
 }
+
+TEST(JSONSchema_frame_2020_12, dynamic_anchor) {
+  const sourcemeta::jsontoolkit::JSON document =
+      sourcemeta::jsontoolkit::parse(R"JSON({
+    "$id": "https://www.sourcemeta.com/schema",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "properties": {
+      "foo": {
+        "$dynamicAnchor": "foo"
+      }
+    }
+  })JSON");
+
+  sourcemeta::jsontoolkit::ReferenceFrame static_frame;
+  sourcemeta::jsontoolkit::ReferenceFrame dynamic_frame;
+  sourcemeta::jsontoolkit::ReferenceMap references;
+  sourcemeta::jsontoolkit::frame(document, static_frame, dynamic_frame,
+                                 references,
+                                 sourcemeta::jsontoolkit::default_schema_walker,
+                                 sourcemeta::jsontoolkit::official_resolver)
+      .wait();
+
+  // Static
+
+  EXPECT_EQ(static_frame.size(), 12);
+  EXPECT_FRAME_2020_12(static_frame, "https://www.sourcemeta.com/schema",
+                       "https://www.sourcemeta.com/schema", "");
+  EXPECT_FRAME_2020_12(static_frame, "https://www.sourcemeta.com/schema#/$id",
+                       "https://www.sourcemeta.com/schema", "/$id");
+  EXPECT_FRAME_2020_12(static_frame,
+                       "https://www.sourcemeta.com/schema#/$schema",
+                       "https://www.sourcemeta.com/schema", "/$schema");
+  EXPECT_FRAME_2020_12(static_frame,
+                       "https://www.sourcemeta.com/schema#/properties",
+                       "https://www.sourcemeta.com/schema", "/properties");
+  EXPECT_FRAME_2020_12(static_frame,
+                       "https://www.sourcemeta.com/schema#/properties/foo",
+                       "https://www.sourcemeta.com/schema", "/properties/foo");
+  EXPECT_FRAME_2020_12(
+      static_frame,
+      "https://www.sourcemeta.com/schema#/properties/foo/$dynamicAnchor",
+      "https://www.sourcemeta.com/schema", "/properties/foo/$dynamicAnchor");
+
+  // JSON Pointers
+
+  EXPECT_FRAME_2020_12(static_frame, "", "https://www.sourcemeta.com/schema",
+                       "");
+  EXPECT_FRAME_2020_12(static_frame, "#/$id",
+                       "https://www.sourcemeta.com/schema", "/$id");
+  EXPECT_FRAME_2020_12(static_frame, "#/$schema",
+                       "https://www.sourcemeta.com/schema", "/$schema");
+  EXPECT_FRAME_2020_12(static_frame, "#/properties",
+                       "https://www.sourcemeta.com/schema", "/properties");
+  EXPECT_FRAME_2020_12(static_frame, "#/properties/foo",
+                       "https://www.sourcemeta.com/schema", "/properties/foo");
+  EXPECT_FRAME_2020_12(static_frame, "#/properties/foo/$dynamicAnchor",
+                       "https://www.sourcemeta.com/schema",
+                       "/properties/foo/$dynamicAnchor");
+
+  // Dynamic
+  EXPECT_EQ(dynamic_frame.size(), 2);
+  // TODO: Should this be here, or should we register it on the static frame?
```

Comment:
Correct. https://www.sourcemeta.com/schema#foo goes to static frame and #foo goes to both